### PR TITLE
fix #84 prevent suite inheritance

### LINKED ISSUE #84: Prevent inheriting from other suites as it is a bad practice
Inheriting from other suites causes the system to run tests twice per derived suite plus one for the inherited suite. So we can consider inheriting from another concrete suite to be a bad practice and we should enforce that by preventing such practice.

### PR DESCRIPTION
**Pull request by silkentrance**
remove extraneous suite inheritance tests which are no longer needed
update documentation
refactor package tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -336,12 +336,9 @@ export abstract class AbstractTestBase {
...
 ```
 
-One can also inherit from another concrete test class or suite, but keep in mind that that the tests declared by that suite will
-be run multiple times and not just in the context of the concrete test classes, see the example provided in `tests/ts/suite.inheritance.suite.ts`.
+Note: You can override test methods inherited from a base class and then call `super()` in order to run the assertions implemented by the super class.
 
-Important: One should not override test methods inherited from a base class and then call `super()` as this will run the tests from the base class twice.
-
-Best practice: Do not inherit from other classes that have been decorated with the ``suite`` decorator. Instead use abstract test base classes.
+Best practice: You must not inherit from other classes that have been decorated with the ``suite`` decorator. Instead use abstract test base classes. Doing so will result in an exception.
```
Comment (pana-cc) [inline README.md]: Reword "Doing so", it is close to the second sentence where the positive case is described. Maybe replace with "Inheriting with classes decorated with ``suite`` will result in an exception."

**Comment by silkentrance** [comment]
@pana-cc with this in place, inheriting from another suite will result in an error. Rationale here is that it is not a good practice to inherit from other suites as this will lead to tests being run multiple times.

File: test.ts [hunk 2]
```diff
@@ -86,43 +87,39 @@ function trimEmptyLines(str: string, eliminateAll = false): string {
...
-    @params({ target: "es5", ts: "pending.v2.suite" })
-    @params({ target: "es6", ts: "pending.v2.suite" })
-    @params({ target: "es5", ts: "only.suite" })
-    @params({ target: "es6", ts: "only.suite" })
-    @params({ target: "es5", ts: "pending.suite" })
-    @params({ target: "es6", ts: "pending.suite" })
-    @params({ target: "es5", ts: "retries.suite" })
-    @params({ target: "es6", ts: "retries.suite" })
-    @params({ target: "es6", ts: "context.suite" })
+    // @params({ target: "es5", ts: "test.suite" })
```
Comment (pana-cc) [inline test.ts]: Are these tests supposed to be commented?

Comment (silkentrance) [reply test.ts]: oh noes...

File: README.md (hunk 1 above)
Comment (silkentrance) [reply README.md]: will do

**Pull request merged**